src/functions/env.ts

```diff
@@ -0,0 +1,203 @@
+  return envStore.vars || {};
+}
+
+/**
+ * Add environment variables to the EnvStore Service.
+ *
+ * @param {Record<string, string>} envs Environment variables to add.
+ * @return {Promise<Record<string, string>>} An object with environment variables from the EnvStore.
+ */
+export async function addEnvs(
+  projectId: string,
+  envs: Record<string, string>
+): Promise<Record<string, string>> {
+  const envStore = await envstore.patchStore(projectId, ENVSTORE_ID, envs);
+  return envStore.vars || {};
+}
+
+/**
+ * Remove given keys from the EnvStore Service.
+ *
+ * @param {string[]} keys Keys of environment variables to remove.
+ * @return {Promise<Record<string, string>>} An object with environment variables from the EnvStore.
+ */
+export async function removeKeys(
+  projectId: string,
+  keys: string[]
+): Promise<Record<string, string>> {
+  const envs: Record<string, string> = {};
+  keys.forEach((key) => {
+    envs[key] = "";
+  });
+  const envStore = await envstore.patchStore(projectId, ENVSTORE_ID, envs);
+  return envStore.vars || {};
+}
+
+/**
+ * Set environment variables in the EnvStore Service to the given set.
+ *
+ * This operation is destructive and deletes env vars not defined in the given
+ * envs.
+ *
+ * @param {Record<string, string>} envs Environment variables to set.
+ * @return {Promise<Record<string, string>>} An object with environment variables from the EnvStore.
+ */
+export async function setEnvs(
+  projectId: string,
+  envs: Record<string, string>
+): Promise<Record<string, string>> {
+  // TODO(taeold): setEnv operation comprises 2 API calls. It isn't an atomic
+  // operation and may leave all env vars to be deleted but not set. setEnv
+  // command is already designed to be destructive, and the outcome of partial
+  // failure isn't too bad (users can simply try the command again). Regardless,
+  // we should work with the EnvStore service team to develop an delete+create
+  // transactionality.
+  await envstore.deleteStore(projectId, ENVSTORE_ID);
+  const envStore = await envstore.createStore(projectId, ENVSTORE_ID, envs);
+  return envStore.vars || {};
+}
+
+/**
+ * Remove all environment variables in the EnvStore Service.
+ *
+ * @return {Promise<Record<string, string>>} An object with environment variables from the EnvStore.
+ */
+export async function clearEnvs(projectId: string): Promise<Record<string, string>> {
+  const envStore = await envstore.deleteStore(projectId, ENVSTORE_ID);
+  return envStore.vars || {};
+}
+
+/**
+ * Clone environment variables in fromProjectId to toProjectId.
+ *
+ * @param {string} fromProjectId Project to clone environment variables from.
+ * @param {string} toProjectId Project to clone environment variables to.
+ * @param {string[]} only List of keys to clone.
+ * @param {string[]} except List of keys to exclude when cloning.
+ * @return {Promise<Record<string, string>>} An object with environment variables from the EnvStore.
+ */
+export async function clone(
```

Comment:
Optional: Given that this has 2 pairs of identically typed args that are _really_ bad to mix up, this is probably a good candidate to use a parameter type

Reply:
Can you expand on what you mean by parameter type? Searching for parameter type on typescript handbook isn't helping me too much.

Reply:
Sorry - I misspoke a bit, the real term is parameter object pattern. Chatted you a link that explains it well, but basically its refactoring this to take a CloneArgs object so it has named parameters